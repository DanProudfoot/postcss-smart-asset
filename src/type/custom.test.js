import { processedCss, compareFixtures } from "../../test/setup"

describe("custom", () => {
  const opts = {
    url(asset, dir, options, decl, warn, result) {
      expect(asset.url).toBeTruthy()
      expect(dir.from).toBeTruthy()
      expect(options).toBeTruthy()
      expect(decl.value).toBeTruthy()
      expect(result.opts).toBeTruthy()
      expect(typeof warn).toBe("function")

      return asset.url.toUpperCase()
    }
  }

  compareFixtures("custom", "should transform url through custom callback", opts)
  compareFixtures("custom", "should transform url through custom callback with array options", [
    { url: "rebase", filter: /\.svg$/ },
    opts
  ])

  compareFixtures("custom-multi", "should transform url through custom callback with multi option", [
    { url: "rebase", filter: /\.svg$/ },
    {
      url(asset) {
        return asset.url.slice(1)
      }
    },
    {
      url(asset) {
        return asset.url.toLowerCase()
      },
      multi: true
    }
  ])
})
