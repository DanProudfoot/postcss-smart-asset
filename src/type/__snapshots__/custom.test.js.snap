// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom should transform url through custom callback 1`] = `
"body {
  background: url(\\"ONE\\"), url('TWO'), url(THREE), url(FOUR), url(/FIVE), url(#SIX);
}"
`;

exports[`custom should transform url through custom callback with array options 1`] = `
"body {
  background: url(\\"ONE\\"), url('TWO'), url(THREE), url(FOUR), url(/FIVE), url(#SIX);
}"
`;

exports[`custom should transform url through custom callback with multi option 1`] = `
"body {
  background: url(\\"ome_url.png\\");
}"
`;
